control-plane/agents/src/bin/core/controller/resources/migration.rs

```diff
@@ -0,0 +1,80 @@
+use serde_json::Value;
+use stor_port::{
+    pstor::{key_prefix_obj, product_v1_key_prefix_obj, StorableObjectType, Store, API_VERSION},
+    types::v0::{
+        store::{definitions::StoreError, pool::PoolSpec, volume::VolumeSpec},
+        transport::PoolTopology,
+    },
+};
+use tracing::{info, warn};
+use utils::{CREATED_BY_KEY, DSP_OPERATOR, MSP_OPERATOR};
+
+/// Migrate the values from legacy keys to latest keys.
+pub async fn migrate_product_v1_to_v2<S: Store>(
+    store: &mut S,
+    spec_type: StorableObjectType,
+) -> Result<(), StoreError> {
+    info!("Migrating {spec_type:?} from v1 to v2 key space");
+    let prefix = &product_v1_key_prefix_obj(spec_type);
+    let store_entries = store.get_values_prefix(prefix).await?;
+    for (k, v) in &store_entries {
+        let id = k
+            .split('/')
+            .last()
+            .ok_or_else(|| StoreError::InvalidKey { key: k.to_string() })?;
+
+        let new_value = match spec_type {
+            StorableObjectType::VolumeSpec => migrate_volume_labels(v.clone())
```

Comment:
can we consume store_entries and skip clone here?

Reply:
Done.